app/Repositories/PostRepository.php

```diff
@@ -95,7 +95,7 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
         // If this is a share-social type post, auto-accept.
         $post->status = $post->type === 'share-social' ? 'accepted' : 'pending';
 
-        $isAdminOrStaff = auth()->user() && auth()->user()->role === 'admin' || auth()->user()->role === 'staff';
+        $isAdminOrStaff = auth()->user() && auth()->user()->role === 'admin' || auth()->user() && auth()->user()->role === 'staff';
```

Comment:
Gah, tricky! It might read nicely to combine these into an `is_array` check:

```php
$isAdminOrStaff = auth()->user() && in_array(auth()->user()->role, ['admin', 'staff'])
```

Reply:
thanks @DFurnes ! I like this way better!